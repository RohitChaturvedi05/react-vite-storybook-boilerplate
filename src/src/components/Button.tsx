import styled, { css } from 'styled-components';

export const Button = styled.button<{
    $type?: 'primary' | 'secondary';
    $displayStyle?: 'outline' | 'iconButton';
    disabled?: boolean;
}>`
    display: flex;
    border: none;
    justify-content: center;
    align-items: center;
    gap: ${(props) => props.theme.spacing.sm};
    border-radius: ${(props) => props.theme.radius.sm};
    cursor: pointer;

    ${({ theme, $type }) => {
        if ($type === 'secondary') {
            return css`
                background-color: ${theme.pallet.secondary.light};
                color: ${theme.pallet.secondary.text};
                &:hover {
                    background-color: ${theme.pallet.secondary.dark};
                }
            `;
        }

        return css`
            background-color: ${theme.pallet.primary.main};
            color: ${theme.pallet.primary.text};
            &:hover {
                background-color: ${theme.pallet.primary.dark};
            }
        `;
    }}

    ${({ $displayStyle }) => {
        if ($displayStyle === 'iconButton') {
            return css`
                padding: 8px;
                border-radius: 100%;
                background-color: transparent;
            `;
        }
    }}

  ${({ theme, disabled }) =>
        disabled &&
        css`
            cursor: not-allowed;
            background-color: ${theme.pallet.secondary.dark};
            color: ${theme.pallet.secondary.text};
            opacity: 0.7;
            &:hover {
                background-color: ${theme.pallet.secondary.dark};
            }
        `}
  [dir="rtl"] & {
        svg {
            transform: rotate(180deg);
        }
    }
`;
