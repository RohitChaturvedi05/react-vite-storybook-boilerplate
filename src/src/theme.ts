const pallet = {
  light: {
    primary: {
      main: "#00a3ff",
      light: "#1aabff",
      dark: "#0081cc",
      text: "#ffffff",
    },
    secondary: {
      main: "#f2f2f2",
      light: "#ffffff",
      dark: "#e6e6e6",
      text: "#595959",
    },
    background: {
      main: "#f2f2f2",
      light: "#ffffff",
      dark: "#e6e6e6",
      text: "#595959",
    },
    text: "#595959",
    divider: "#d9d9d9",
  },
  dark: {
    primary: {
      main: "#00a3ff",
      light: "#1aabff",
      dark: "#0081cc",
      text: "#ffffff",
    },
    secondary: {
      main: "#222831",
      light: "#ffffff",
      dark: "#e6e6e6",
      text: "#595959",
    },
    background: {
      main: "#1a1a1a",
      light: "#323232",
      dark: "#0d0d0d",
      text: "#ffffff",
    },
    text: "#595959",
    divider: "#404040",
  },
};

const theme = {
  spacing: {
    xs: "4px",
    sm: "8px",
    md: "16px",
    lg: "24px",
    xl: "32px",
    xxl: "40px",
  },
  pallet: {
    primary: {
      main: "#00a3ff",
      light: "#00a3ff",
      dark: "#00a3ff",
      text: "#ffffff",
    },
    secondary: {
      main: "#f2f2f2",
      light: "#ffffff",
      dark: "#e6e6e6",
      text: "#595959",
    },
    background: {
      main: "#f2f2f2",
      light: "#ffffff",
      dark: "#e6e6e6",
      text: "#595959",
    },
    text: "#595959",
    divider: "#d9d9d9",
  },
  radius: {
    sm: "8px",
    md: "12px",
    circle: "100%",
  },
  fontSize: {
    xxSmall: "xx-small",
    xSmall: "x-small",
    small: "small",
    medium: "medium",
    large: "large",
    larger: "larger",
    xLarge: "x-large",
    xxLarge: "xx-large",
    xxxLarge: "xxx-large",
  },
};

export const createTheme = (mode: "light" | "dark" = "light") => {
  return {
    ...theme,
    pallet: pallet[mode],
  };
};
