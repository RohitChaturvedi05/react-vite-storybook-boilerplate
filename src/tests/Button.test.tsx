import React from 'react';
import { describe, expect, it } from 'vitest';
import { Button } from '../src/components/Button';
import { createTheme } from '../src/theme';
import { renderWithTheme } from './utils';

describe('Button component', () => {
    it('Button should render correctly', () => {
        const theme = createTheme('light');
        const { container } = renderWithTheme(<Button>Button</Button>);
        expect(container).toMatchSnapshot();
    });
});
