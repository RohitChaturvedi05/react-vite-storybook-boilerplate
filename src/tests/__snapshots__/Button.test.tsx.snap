// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > Button should render correctly 1`] = `
<div>
  <button
    class="sc-blHHSb bPhOSb"
  >
    Button
  </button>
</div>
`;
